[WIP] Cybersun Ship rebalance (actually nerf).

EgorDinamit opened the pull request:
## About The Pull Request

**Rebalances loot on the ship, adds 2 random loot spawners.**

Amount of materials they get round-start is lowered by A LOT.
2 boxes with Tier4 parts - replaced with one box containing T1 parts for protolathe, autolathe and circuit imprinter.
Adds 2 types of this ruin that doesn't contain ghost-role spawners and can't spawn with each other.
It's done to provide a solid 30% chance for ghost-role one to spawn.

## Why It's Good For The Game

Given the number of materials you get on the ship right now, you can easily finish almost every machine at 20 minutes mark.
With this update, you won't be able to simply live on the ship and use your round-start mats for 2 hours, you'll have to, at very least, mine ores around the ship and probably on asteroids far away.

Probably, this will make them actually explore space, at least to get materials.
Basically it just means no more 30-minutes mechas, cyborg armies and so on.

## Changelog
:cl: EgorDinamit
add: 2 subtypes of forgottenship ruin. Both of those don't contain ghost-role spawners. Only one type of ruin can spawn at the same time, providing a solid 30% chance for ghost-role type to spawn.
balance: Cybersun Ship now contains a lower amount of materials, R&D crate moved to high-sec part, random "gimmick" crate added: may contain useful, or useless stuff.
/:cl:

EgorDinamit committed:
Update ghost_role_spawners.dm

EgorDinamit committed:
Update ghost_role_spawners.dm

EgorDinamit committed:
Update ghost_role_spawners.dm

EgorDinamit committed:
Update ghost_role_spawners.dm

EgorDinamit committed:
Removing all copy-pasta

EgorDinamit committed:
Main Commit

ExcessiveUseOfCobblestone commented on code/game/objects/items/storage/boxes.dm:
```diff
@@ -1174,6 +1174,18 @@
+	var/static/items_inside = list(
+		/obj/item/stock_parts/matter_bin/bluespace = 6,
+		/obj/item/stock_parts/manipulator/femto = 5,
+		/obj/item/reagent_containers/glass/beaker = 4)
```
what's with the T4 parts?

EgorDinamit replied:
It's using T4 parts right now, I just reduced the amount of them.

ExcessiveUseOfCobblestone replied:
WHAT IS THIS RUIN 😠

EgorDinamit replied:
Alright alright, it always had Deluxe stock parts boxes, same with lavaland base.
I'm okay with lowering the tier of parts, but not to the first tier.

Qustinnus replied:
make it first tier

EgorDinamit replied:
@Qustinnus 
No.

Qustinnus replied:
then ill open a pr myself 😏

EgorDinamit replied:
I may see a reason to lower the tier by a bit, but I don't see any reasons to set it to the lowest one.

ExcessiveUseOfCobblestone commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -446,6 +446,73 @@
+				/obj/item/circuitboard/machine/processor/slime,
+				/obj/item/circuitboard/machine/monkey_recycler,
+				/obj/item/circuitboard/computer/xenobiology,
+				/obj/item/stock_parts/matter_bin/bluespace,
+				/obj/item/stock_parts/matter_bin/bluespace,
+				/obj/item/stock_parts/manipulator/pico,
+				/obj/item/stock_parts/manipulator/pico,
```
T4/T3 parts?

EgorDinamit replied:
I probably can move it 1 tier down.

Qustinnus commented on code/game/objects/structures/crates_lockers/crates/secure.dm:
```diff
@@ -49,6 +49,71 @@
+	var/max_c = 50 //Maximum amount.
+	var/min_r = 5  //Ditto for rares
+	var/max_r = 15
+	var/pickednum = rand(1, 50)
```
this should use a weighted list instead

EgorDinamit committed:
One tier down.

Lowered tier of stock parts.

ExcessiveUseOfCobblestone commented:
Can you please put what you want out of the role [here](https://hackmd.io/mPfrb52gSYmDe8wsgvq9zA?both) so I know where I need to tell you to tone it down before it gets to the point of no return. If you can't access edit ability you may need to ask oranges for perms via discord.

The more I read about this role and see what all they get the more it comes off as a codebaby that keeps getting fed without consideration of either how it plays ingame or without consideration of the rest of the game entirely. I'd like to have it toned down to fit with what you had perhaps initially designed the role to be (or we shift it to a specific focus if you did not have anything in mind) before I just start looking into full removal (assuming admins dont config deactivate it first).

SumFaggotPlayTester commented:
Can you also throw in a lower chance for it to spawn? This is a good ruin in of itself but I'm starting to see it every round. This balance is nice as well given how much I've fiddled with it given the current meta for it gets all the machines up & running way to quickly. The idea where you fly around spess looking for mats for the ship seems like the best place this can go as you'll have hostile sapient people fighting/avoiding/sabatoging spess explorers looking for similar loot.

Tlaltecuhtli commented:
remove the minerals, if they want them they can go find withe ship with gps to go to lavaland or mine asteroids  on the way, no need for free shit that not even golems get

EgorDinamit commented:
@SumFaggotPlayTester I'd need to change chance for every space ruin to do so.
Though I can, instead, add 1-2 ruins that won't spawn with it to lower the chance.

Cartographer-D commented:
> @SumFaggotPlayTester Their main objective, gameplay-wise, is **creating a sense of threat** for >anyone who ended up **in space**, probably the same as ashwalkers for lavaland, but while >ashwalkers are technically infinite, in space you might have only 5 humans and somewhat limited >amount of cyborgs, they might create.

So if you want a sense of threat in space, here's what you can do:
A. Add a navigation console
B. Give them a reason to take from other people (Item Analyzer, DNA analyzer, objectives)
C. Make the players the threat and not their OP turrets.
D. Limit their resources so they have to leech off of others.


> @SumFaggotPlayTesterWhile in my opinion they should be allowed to **visit** the station for any >sort of **minor** antagonistic actions(such as stealing materials, non high-risk items), they mustn't >serve a role of main antagonist in any sense.
>This means they shouldn't try to kill anyone on the station, especially with no real reason. 
You say that you want them to visit without killing anybody, but their weapons are as follows:
1. Pistol with ammo
2. A combat shotgun
3. Anything they can print in the lathe.
4. A Syndicate AI module

If you don't want them to be lethal, you should supply them with non-lethal weaponry.

Also, visiting the station as antags gives them the opportunity to escalate. If someone is captured by the station, do you think players will just say "okay, they're a goner"? No. I would say that if you want them to interact just have them be a ghost role, but then they are basically what pirates are already.

My suggestion would be to double down on what makes them interesting: They are a threat in space... ONLY. This keeps them from using a ghost role to interfere with the station (which isn't encouraged) without removing the main purpose you stated. How do you do this?
A. Prevent them from being able to teleport at all or navigate via console to the station.
B. Add in their flavor text that they shouldn't go to the station.
C. Remove the AI

> @SumFaggotPlayTesterWhile They are supposed to become a serious threat to almost anyone at >40-minutes mark minimum, but at the same time, their behavior should be limited to "maximum >defense, minimum offense".

What this will amount to is a bunch of players with a lot of deadly equipment seeking a reason to use it. Defense can mean a lot of things, and I've seen players use "defense" as a means to pre-emptively bomb a roboticist making a gygax among other things.

If you really want them to be able to impact a round, maybe their actions can somehow result in support for traitors on the station (dropping them gear, generating TC, temporarily disabling the powernet, things like that.)

TerraGS commented:
I'd be in favor of a bigger time gate to reaching the point of being a reasonable threat to the station. Would also like if it spawned less often to make it more of a special thing that you can't expect to be around on a regular basis and to prevent degeneracy.

EgorDinamit commented:
@TerraGS, I can try to add 2 types of this ruin, tha can't spawn with each other giving you a somewhat stable 33% chance to get this specific ruin.

@Cartographer-D, by navigation console, do you mean making it the working shuttle?

Turrets are weak for anyone who got a guns, bombs or tools to avoid them.

This PR is limiting their resources enough already, but does it randomly, so you might end up with no resources at all, or a big amount of every resource.

I can't remove their ability to create teleporter and AIs, and even if I did some shit-coding to add some sort of brain trauma, I don't want to nerf their abilities into stone age.

Flavor text supports policy, it's up to headmins to edit it any time.

Teleporter also exists to help traitors, as you said, but generating TCs sounds like a questionable idea, but I would like to give them some sort of unique item that will be useful for anyone who isn't on the ship.

senjx commented:
BASED ORANGES

Shadowflame909 commented:
Remove the lethals. Give them CQC/E-Bow and a stealth implant.

Yeah, kind of full MGS parody. But it's pretty much better then incentivizing death and doom.

RaveRadbury commented:
> being a reasonable threat to the station. 

The thing is that they shouldn't be a threat to the station. This is a scavenger ghost role, not nukies lite.

EgorDinamit commented:
@RaveRadbury by "Threat to the station" I pretty much meant station's crew in general.
I have to balance it to be able to win in fair fight, but lose in full sec assault, if they are not prepared.

I rarely see station actually raid the ship, I may try and add me sort of **equivalent** of "malf AI disk" - something almost useless for ghost role itself, but unique and useful for the station.

RaveRadbury commented:
@EgorDinamit Raids definitely happen, and the flavortext describes them as needing to fort and get supplies. Admins have been ruling according to the flavortext.

If you're intending for them to raid the station this is drastically different from what we expect from ghost role antags. Given that it is rolling every round, this is nukies lite then, and runs a strong risk of disrupting _any_ round given that it is a _guaranteed spawn_.

EgorDinamit commented:
@RaveRadbury they aren't allowed to RAID the station, but due to headmins' policy they are allowed to steal some stuff, again, without killing indiscriminately.

I will try to fix guaranteed spawn problem in this PR.

I said _**equivalent**_ of this disk. I need something ultimately useless for the ghost role, but useful for the station. 
They already spawn with syndicate documents, but this thing is useful for cargo only.

RaveRadbury commented:
> I said _**equivalent**_ of this disk. I need something ultimately useless for the ghost role, but useful for the station.

Do you mean the combat upgrade module? I thought you meant like a law 0 antag for AI.

EgorDinamit commented:
Yes, ashwalkers get this combat upgrade disk which is quite useless for them.
I need something useless for cybersun crew and useful for the station.

EgorDinamit committed:
Removing 100% chance for ruin to spawn.

Okay, that may look strange, but I added 2 more ruins which are somewhat useless versions of former ghost-role ruin. Only one type of it can spawn at the same time andd therefore you won't see this ghost-role every shift.

EgorDinamit committed:
MapMerge

FloranOtten commented:
If you're giving them both a syndicate law module, and a combat upgrade, you've effectively given them the ability to ally a malf AI. That's quite a powerful thing.

EgorDinamit commented:
Can you please read what I'm writting for once?
I'm not giving them a goddamn malf ai upgrade, read again before you understand.

FloranOtten commented:
>Do you mean the combat upgrade module?

>Yes, 

Well what the fuck DO you mean?

EgorDinamit commented:
@FloranOtten 

> I rarely see station actually raid the ship, I may try and add me sort of **_equivalent_** of "malf AI disk" - _something almost useless for ghost role itself, but unique and useful for the station._

**_Equivalent._**

EgorDinamit committed:
Merge branch 'policytext' into forgottenshipfix3

EgorDinamit committed:
Minor fixes + Fluff antag datum

Antag datum needed for admemery.

EgorDinamit committed:
Merge branch 'master' into forgottenshipfix3

EgorDinamit committed:
Update sql_ban_system.dm

EgorDinamit committed:
Ammo Type fixes

EgorDinamit committed:
No more T3 parts.

Actually, no parts for xenobio-kit.

EgorDinamit committed:
Lowest tier for parts.

"We're sorry to say that, but all stock parts we had in store went to our lavaland base so we had to buy stock parts from Nanotrasen."
"Yes, we do understand that Nanotrasen has the shittiest tech in the universe, but we had no choice!"

ExcessiveUseOfCobblestone commented:
DNM for 2 weeks.

Qustinnus commented:
@ExcessiveUseOfCobblestone so are you doing anything as an alternative like removing it?

ExcessiveUseOfCobblestone commented:
I didn't have any plans on personally making changes to the ruin, no. why do you ask? @Qustinnus
